src/ExtranetBundle/objects/db/TDataExtranetUser.class.php

```diff
@@ -649,8 +649,11 @@ public function Logout()
                         $new[$key] = $all[$key];
                     }
                 }
-                $_SESSION = array();
-                $session->replace($new);
+
+                $session->invalidate(); // also clears $_SESSION
+                if (\count($new) > 0) {
+                    $session->replace($new);
+                }
```

Comment:
Can you explain, why this solves the problem? I am missing some context. A small business summary would be helpful. :)

Reply:
The "invalidate" is the important change here: replacing the session via SessionInterface::replace does not directly affect the $_SESSION variable and it is out of sync then. However other code in chameleon-base accesses that variable directly.

invalidate mitigates that problem.